### PR TITLE
Data collector improvements

### PR DESCRIPTION
**Pull request by fdalcin**
Issue: #2370 . Accompanies pressbooks/pressbooks-network-analytics#155

This PR adds new information to the data collectors

#### Site metada
- New site meta informing which users are network managers -- a serialized value exists, however a new comma separated value was added to be used by the BI dashboard.

#### Book data collector
- Include information about hypothesis enabled

#### User data collector
- Highest user role
- Total books
- Books as admin
- Books as editor
- Books as author
- Books as contributor/collaborator
- Books as subscriber

#### Book data collector

1. Go to Settings > Hypothesis and select one or more of the following: Allow on parts, Allow on chapters, Allow on front matter, Allow on back matter
2. Save the book info again to trigger data collector
3. Check `wp_blogmeta` table for `pb_hypothesis_activated` meta key -- it should be active for books that allow at least one of the four options

#### Network managers

1. In the Network root go to Settings > Network Managers and add/remove a user as network manager. 
2. Check `wp_sitemeta` table -- it should now have a new meta key called `pressbooks_network_managers_ids` with a comma separated list of network managers. Note that the `pressbooks_network_managers` is still present and being used. 

#### User data collector 
1. Setup multiple users with roles in different books
2. Check `wp_usermeta` table for any meta key starting with `pb_`
3. Edit one of the users profile
4. Check `wp_usermeta` table for any meta key starting with `pb_` -- it should contain the following new keys `pb_highest_role`, `pb_total_books`, `pb_books_as_admin`, `pb_books_as_editor`, `pb_books_as_author`, `pb_books_as_contributor`, `pb_books_as_subscriber`

If and when this PR gets merged, it might be a good thing to improve the [Network Analytics Plugin](https://github.com/pressbooks/pressbooks-network-analytics).

### CODE REVIEW

File: inc/datacollector/class-book.php [hunk 1]
```diff
@@ -104,13 +118,6 @@ static public function hooks( Book $obj ) {
 		add_action( 'wp_delete_site', [ $obj, 'deleteSite' ], 999 );
 	}
 
-	/**
-	 *
-	 */
-	public function __construct() {
-
-	}
-
```
Comment (fdalcin) [inline inc/datacollector/class-book.php:113]: Not needed

File: inc/datacollector/class-book.php [hunk 2]
```diff
@@ -455,7 +467,7 @@ public function getTotalNetworkStorageBytes() {
 	 */
 	public function getTotalBooks() {
 		global $wpdb;
-		$root_id = 1; // root network id should not be considered
+		$root_id = get_network()->site_id; // root network id should not be considered
```
Comment (fdalcin) [inline inc/datacollector/class-book.php:470]: This is a better and reliable way to check the root network id

File: inc/admin/networkmanagers/namespace.php [hunk 3]
```diff
@@ -88,8 +88,11 @@ function update_admin_status() {
 
 		if ( is_array( $restricted ) && ! empty( $restricted ) ) {
 			update_site_option( 'pressbooks_network_managers', $restricted );
+			// This meta option will only be used by the BI, but we want to keep that in sync.
+			update_site_option( 'pressbooks_network_managers_ids', implode( ',', $restricted ) );
```
Comment (fdalcin) [inline inc/admin/networkmanagers/namespace.php:92]: New meta key for network managers information

File: inc/datacollector/class-book.php [hunk 4]
```diff
@@ -306,6 +313,11 @@ public function copyBookMetaIntoSiteTable( $book_id ) {
 		$wp_quicklatex_activated = is_plugin_active_for_network( 'wp-quicklatex/wp-quicklatex.php' ) || is_plugin_active( 'wp-quicklatex/wp-quicklatex.php' );
 		update_site_meta( $book_id, self::WP_QUICK_LATEX_ACTIVATED, $wp_quicklatex_activated ? 1 : 0 );
 
+		$hypothesis_options = get_option( 'wp_hypothesis_options' ) ?: [];
+		$active_options = array_intersect_key( array_flip( self::$hypothesis_keys ), $hypothesis_options );
+
+		update_site_meta( $book_id, self::HYPOTHESIS_ACTIVATED, empty( $active_options ) ? 0 : 1 );
+
```
Comment (fdalcin) [inline inc/datacollector/class-book.php:320]: New Hypothesis book metadata

File: inc/datacollector/class-user.php [hunk 5]
```diff
@@ -61,4 +171,53 @@ public function setSubscriberRole( $user_login, $user ) {
...
+	private function getHighestRole( array $roles ) {
+		// Get all roles that the user is associated.
+		$all_roles = array_filter(
+			$roles, function ( $value ) {
+				return $value;
+			}
+		);
+
+		// Get the highest score based on the user roles.
+		$highest_score = array_reduce(
```
Comment (arzola) [inline inc/datacollector/class-user.php:202]: Nice

**Review by arzola — APPROVED** [review]
LGTM Nice job, btw I could't see the `pressbooks_network_managers_ids` populated maybe I'm doing something wrong

**Comment by fdalcin** [comment]
Do you have users other than the admin one to set as Network Manager? If not, try granting super admin privileges to other uses and then adding them as Network Managers

**Pull request merged**